### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by FedeOt**
-Adds registration component
-Adds unit tests

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -39,6 +37,21 @@ export const updateAccount = (id, name) =>
   );
 
 
+export const createNewUser = (user) =>
+  axiosClient.post("/user", user, {
+    headers: { Authorization: getAuthHeader() },
+  });
+
+export const addRoleApi = (id) =>
```
Comment (ghost) [inline src/api/index.js]: _addApiRole_ sounds better to me!

File: src/components/Registration.jsx [hunk 2]
```diff
@@ -0,0 +1,323 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { addRoleApi, createNewUser } from "../api";
+import { ValidateFields } from "../helpers/validateRegisterFields";
```
Comment (ghost) [inline src/components/Registration.jsx]: The name of this function should follow the _camelCase_ naming convention.

File: src/components/Registration.jsx [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+import { InputText } from "./InputText";
+import { InvalidField } from "./InvalidField";
```
Comment (ghost) [inline src/components/Registration.jsx]: This name may be confusing. You can probably rename it to _InvalidFieldMessage_ or something like that.

File: src/components/Registration.jsx [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+  lastName: "",
+  locality: "",
+  mobilePhone: "",
+  password: "",
+  postalCode: "",
+  region: "",
+  ssn: "",
+  title: "Mr.",
+  workPhone: "",
+};
```
Comment (ghost) [inline src/components/Registration.jsx]: You can put the `initialState` object in a `data` folder and import it here.

File: src/moment/validateDate.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import moment from "moment/moment";
```
Comment (ghost) [inline src/moment/validateDate.js]: I believe `import moment from 'moment'` works too.

File: src/pages/MainScreen.jsx [hunk 6]
```diff
@@ -18,6 +19,7 @@ export const MainScreen = () => {
           <Routes>
             <Route path='accounts' element={<CheckingAccountsList/>}/>
             <Route path='creation' element={<CheckingAccountCreation/>}/>
+            <Route path='registration' element={<Registration/>}/>
```
Comment (ghost) [inline src/pages/MainScreen.jsx]: These are common path examples you can use:

/checking-account/create

/user/create

File: src/components/InvalidField.jsx [hunk 7]
```diff
@@ -0,0 +1,7 @@
+
+
+export const InvalidField = (props) => {
```
Comment (ghost) [inline src/components/InvalidField.jsx]: _InvalidFieldMessage_ sounds like a more appropriate name for this component.

File: src/helpers/validateRegisterFields.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    if(data.homePhone.trim() === ''){
+        errors.homePhone = true; 
+    }
+    if(data.postalCode.trim() === ''){
+        errors.postalCode = true; 
+    }
+    if(data.country.trim() === ''){
+        errors.country = true; 
+    }
```
Comment (ghost) [inline src/helpers/validateRegisterFields.js]: If you add the `required` attribute to the corresponding elements, you can get rid of repeated validations and just leave Email and SSN validations.

Comment (ghost) [reply src/helpers/validateRegisterFields.js]: Also, you don't have to worry about empty field messages. The `required` attribute does that for you :)

Screen Shot 2022-10-19 at 4 17 59 PM

File: src/tests/UserRegistration.test.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    locality: "SomeLocality",
+    mobilePhone: "99999999",
+    password: "demo123!",
+    postalCode: "33101",
+    region: "SomeRegion",
+    ssn: "457-88-9677",
+    workPhone: "99999999",
+  };
+
+test('Should redirect to the login after create a new user', async()=>{
```
Comment (ghost) [inline src/tests/UserRegistration.test.js]: ... after creating...*

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/api/index.js (hunk 1 above)
Comment (FedeOt) [reply src/api/index.js]: Done!

File: src/components/Registration.jsx (hunk 2 above)
Comment (FedeOt) [reply src/components/Registration.jsx]: Done!

File: src/components/Registration.jsx (hunk 3 above)
Comment (FedeOt) [reply src/components/Registration.jsx]: Done!

File: src/components/Registration.jsx (hunk 4 above)
Comment (FedeOt) [reply src/components/Registration.jsx]: Done!

File: src/moment/validateDate.js (hunk 5 above)
Comment (FedeOt) [reply src/moment/validateDate.js]: Done!

File: src/components/InvalidField.jsx (hunk 7 above)
Comment (FedeOt) [reply src/components/InvalidField.jsx]: Done!

File: src/helpers/validateRegisterFields.js (hunk 8 above)
Comment (FedeOt) [reply src/helpers/validateRegisterFields.js]: Would that prevent the user from enter empty spaces?

File: src/tests/UserRegistration.test.js (hunk 9 above)
Comment (FedeOt) [reply src/tests/UserRegistration.test.js]: right!

File: src/helpers/validateRegisterFields.js (hunk 8 above)
Comment (ghost) [reply src/helpers/validateRegisterFields.js]: We talked about it through Slack!

**Review by ghost — APPROVED** [review]

**Pull request merged**